app/src/backend/environmentVariables/index.ts

```diff
@@ -0,0 +1,68 @@
+import firebaseApp from "firebase";
+import { deleteField, doc, getFirestore, onSnapshot, setDoc, updateDoc } from "firebase/firestore";
+import { EnvironmentVariable } from "./types";
+
+const getDocPath = (ownerId: string) => {
+  const db = getFirestore(firebaseApp);
+  return doc(db, "environmentVariables", ownerId);
+};
+
+export const setEnvironmentVariablesInDB = async (
+  ownerId: string,
+  payload: {
+    newVariables: EnvironmentVariable;
+    environment: string;
+  }
+) => {
+  const newVariables = Object.fromEntries(
+    Object.entries(payload.newVariables).map(([key, value]) => [key, { syncValue: value.syncValue, type: value.type }])
+  );
+
+  return setDoc(
+    getDocPath(ownerId),
+    {
+      [payload.environment]: {
+        ...newVariables,
+      },
+    },
+    { merge: true }
+  );
+};
+
```

Comment:
Each environment should be a separate doc in the collection `envs`

`environmentVariables/{ownerId}/envs/{envName}`

```
name: EnvName,
variables: {

}
```